File: api/models/report.py
"""Report data models for coach analysis reports."""

from pydantic import BaseModel
from typing import Literal


class Strength(BaseModel):
    """A strength/positive point from the analysis."""
    text: str
    frame_refs: list[str]  # ["Frame 1", "Frame 2"]


class Issue(BaseModel):
    """An issue found in the analysis."""
    text: str
    frame_refs: list[str]
    severity: Literal["high", "medium", "low"] = "medium"


class ImprovementSuggestion(BaseModel):
    """A numbered improvement suggestion."""
    text: str
    number: int


class IssueSummaryRow(BaseModel):
    """A row in the issue summary table from frames.json data."""
    frame_number: int
    issue_type: str
    body_part: str
    description: str
    priority: Literal["high", "medium", "low"]


class RadarScores(BaseModel):
    """6-dimension radar scores (1-10 each)."""
    hitting_technique: int = 5
    footwork: int = 5
    body_rotation: int = 5
    timing: int = 5
    fitness: int = 5
    tactics: int = 5


class CoachReport(BaseModel):
    """Complete structured coach analysis report."""
    video_id: str
    overall_score: int  # 1-10
    radar_scores: RadarScores
    strengths: list[Strength]
    issues: list[Issue]
    improvement_suggestions: list[ImprovementSuggestion]
    issue_summary: list[IssueSummaryRow]
    coach_summary: str
    training_plan: str  # 专项训练计划

File: web/src/api/report.ts
import axios from 'axios'

const API_BASE = import.meta.env.VITE_API_BASE_URL || 'http://localhost:8000'

export interface Strength {
  text: string
  frame_refs: string[]
}

export interface Issue {
  text: string
  frame_refs: string[]
  severity: 'high' | 'medium' | 'low'
}

export interface ImprovementSuggestion {
  text: string
  number: number
}

export interface IssueSummaryRow {
  frame_number: number
  issue_type: string
  body_part: string
  description: string
  priority: 'high' | 'medium' | 'low'
}

export interface RadarScores {
  hitting_technique: number
  footwork: number
  body_rotation: number
  timing: number
  fitness: number
  tactics: number
}

export interface CoachReport {
  video_id: string
  overall_score: number
  radar_scores: RadarScores
  strengths: Strength[]
  issues: Issue[]
  improvement_suggestions: ImprovementSuggestion[]
  issue_summary: IssueSummaryRow[]
  coach_summary: string
  training_plan: string
}

export async function getCoachReport(videoId: string): Promise<CoachReport> {
  const response = await axios.get<CoachReport>(`${API_BASE}/api/report/${videoId}`)
  return response.data
}